core/src/test/scala/org/broadinstitute/dsde/rawls/util/LibraryPermissionsSpec.scala

```diff
@@ -0,0 +1,194 @@
+package org.broadinstitute.dsde.rawls.util
+
+import java.util.UUID
+
+import org.broadinstitute.dsde.rawls.{RawlsException, RawlsExceptionWithErrorReport}
+import org.broadinstitute.dsde.rawls.dataaccess.MockGoogleServicesDAO
+import org.broadinstitute.dsde.rawls.model.{AttributeName, UserInfo, Workspace, WorkspaceAccessLevels}
+import org.scalatest.FreeSpecLike
+import slick.dbio.{DBIO, SuccessAction}
+import org.joda.time.DateTime
+import spray.http.OAuth2BearerToken
+
+import scala.concurrent.ExecutionContext
+
+/**
+  * Created by ahaessly on 3/31/17.
+  */
+class LibraryPermissionsSpec extends FreeSpecLike with LibraryPermissionsSupport {
+  implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
+  val gcsDAO: MockGoogleServicesDAO = new MockGoogleServicesDAO("test")
+  val userInfo: UserInfo = UserInfo("owner-access", OAuth2BearerToken("token"), 123, "123456789876543212345")
+
+  val discoverabilityOnly = Seq(discoverableWSAttribute)
+  val publishName = Seq(publishedFlag)
+  val publishNames = Seq(publishedFlag, AttributeName("library", "anothertest"))
+  val modifyNames = Seq(AttributeName("library", "test2"), AttributeName("library", "anothertest"))
+  val modifyName = Seq(AttributeName("library", "test2"))
+
+  val testWorkspace = new Workspace(workspaceId=UUID.randomUUID().toString,
+    namespace="testWorkspaceNamespace",
+    name="testWorkspaceName",
+    realm=None,
+    isLocked=false,
+    createdBy="createdBy",
+    createdDate=DateTime.now(),
+    lastModified=DateTime.now(),
+    attributes=Map.empty,
+    bucketName="bucketName",
+    accessLevels=Map.empty,
+    realmACLs=Map())
+
+
+  "LibraryPermissionsSupport" - {
```

Comment:
I don't think these tests are sufficient because they all rely on direct access of getPermissionChecker but none test if getPermissionChecker is called correctly. I can't really tell from the rest of the tests whether or not all cases are tested. I would rather see either test data constructed that exercises all cases or mock out the calls to dataAccess.workspaceQuery inside the for comprehension of withLibraryPermissions.